src/main/scala/mesosphere/mesos/TaskGroupBuilder.scala

```diff
@@ -342,14 +341,20 @@ object TaskGroupBuilder {
 
   private[this] def computeHealthCheck(
     healthCheck: raml.HealthCheck,
-    endpoints: Seq[raml.Endpoint]): mesos.HealthCheck.Builder = {
+    podDefinition: PodDefinition,
+    hostPorts: Seq[Option[Int]]): mesos.HealthCheck.Builder = {
+
     val builder = mesos.HealthCheck.newBuilder
     builder.setDelaySeconds(healthCheck.delaySeconds.toDouble)
     builder.setGracePeriodSeconds(healthCheck.gracePeriodSeconds.toDouble)
     builder.setIntervalSeconds(healthCheck.intervalSeconds.toDouble)
     builder.setConsecutiveFailures(healthCheck.maxConsecutiveFailures)
     builder.setTimeoutSeconds(healthCheck.timeoutSeconds.toDouble)
 
+    lazy val hostPortsByEndpoint: Map[String, Option[Int]] = {
```

Comment:
worth an `assume` to check that hostPorts and the collection of endpoint names are the same size?